src/manifolds/Hamiltonian.jl

```diff
@@ -0,0 +1,211 @@
+#
+# This file requires Symplectic to be defined, since it need the symplectic inverse A^+
+# This type is used in the Symplectic Grassmann
+@doc raw"""
+    Hamiltonian{T,S<:AbstractMatrix{<:T}} <: AbstractMatrix{T}
+
+A type to store an hamiltonien matrix, i.e. A square matrix matrix for which ``A^+ = -A`` where
+
+```math
+A^+ = J_{2n}A^{\mathrm{T}}J_{2n}, \qquad J_{2n} \begin{pmatrix} 0 & I_n\\-I_n & 0 \end{pmatrix},
+```
+
+and ``I_n`` denotes the ``n×n``
+"""
+struct Hamiltonian{T,S<:AbstractMatrix{<:T}} <: AbstractMatrix{T}
+    value::S
+    function Hamiltonian(A::S) where {T,S<:AbstractMatrix{<:T}}
+        n = div(size(A, 1), 2)
+        @assert size(A, 1) == 2 * n "The first dimension of A ($(size(A,1))) is not even"
+        @assert size(A, 2) == 2 * n "The matrix A is of size ($(size(A))), which is not square."
+        return new{T,S}(A)
+    end
+end
+# Avoid double wrapping / unwrap if that happened
+Hamiltonian(A::Hamiltonian) = Hamiltonian(A.value)
+# Conversion
+function Matrix(A::Hamiltonian)
+    return Matrix(A.value)
+end
+
+Base.:*(H::Hamiltonian, K::Hamiltonian) = Hamiltonian(H.value * K.value)
+Base.:+(H::Hamiltonian, K::Hamiltonian) = Hamiltonian(H.value .+ K.value)
+Base.:-(H::Hamiltonian, K::Hamiltonian) = Hamiltonian(H.value .- K.value)
+
+function show(io::IO, A::Hamiltonian)
+    return print(io, "Hamiltonian($(A.value))")
+end
+size(A::Hamiltonian) = size(A.value)
+
+@doc raw"""
+    HamiltonianMatrices{T,𝔽} <: AbstractDecoratorManifold{𝔽}
+
+The [`AbstractManifold`](https://juliamanifolds.github.io/ManifoldsBase.jl/stable/types.html#ManifoldsBase.AbstractManifold)
+consisting of (real-valued) hamiltonian matrices of size ``n×n``, i.e. the set
+
+````math
+\mathfrak{sp}(2n,𝔽) = \bigl\{p  ∈ 𝔽^{2n×2n}\ \big|\ p^+ = p \bigr\},
+````
+where ``⋅^{+}`` denotes the [`symplectic_inverse`](@ref),. and ``𝔽 ∈ \{ ℝ, ℂ\}``.
+
+Though it is slightly redundant, usually the matrices are stored as ``2n×2n`` arrays.
+
+The symbol refers to the main usage within `Manifolds.jl` that is the
```

Comment:
This is unclear. What symbol do you mean?

Reply:
the mathfrak one, one could add The symbol ``\mathfak{sp}`` here. (Hamiltonian itself has not that many s or p in it).